include/nil/blueprint/parser.hpp

```diff
@@ -449,7 +458,8 @@ namespace nil {
                     }
                     case llvm::Instruction::ICmp: {
                         auto cmp_inst = llvm::cast<const llvm::ICmpInst>(inst);
-                        if (cmp_inst->getOperand(0)->getType()->isIntegerTy())
+                        if (cmp_inst->getOperand(0)->getType()->isIntegerTy()
+                            || cmp_inst->getOperand(0)->getType()->isFieldTy())
                             handle_int_cmp(cmp_inst, variables);
                         else if (cmp_inst->getOperand(0)->getType()->isPointerTy())
```

Comment:
Could you please add `else` branch with some kind of panic?

Reply:
Sure. What kind of panic is better, than `assert`, which causes a lot of confusion in release builds?

Reply:
Done